source/swap/test/Swap-unit.js

```diff
@@ -134,21 +138,27 @@ contract('Swap Unit Tests', async accounts => {
 
     it('test adding token that does not transfer swap incorrectly and transfer returns false', async () => {
       // create mocked contract to test transfer
-      const fungibleTokenTemplate = await FungibleToken.new()
+      const handlerTemplate = await ERC20TransferHandler.new()
       const tokenMock = await MockContract.new()
+      const TransferHandlerRegistryTemplate = await TransferHandlerRegistry.new()
+      const FungibleTokenTemplate = await FungibleToken.new()
 
-      const token_balance = fungibleTokenTemplate.contract.methods
-        .balanceOf(EMPTY_ADDRESS)
+      const getHandler = TransferHandlerRegistryTemplate.contract.methods
+        .getTransferHandler(kind)
         .encodeABI()
 
-      const token_transfer = fungibleTokenTemplate.contract.methods
+      await mockRegistry.givenMethodReturnAddress(
+        getHandler,
+        handlerTemplate.address
```

Comment:
I'm afraid this test hasn't been mocked properly.
The call within `Swap.swap` performs as follows (with the current mocking):
- `mockRegistry. getHandler()`
- returns: `handlerTemplate.address` - as mocked on line 150
- swap performs `handlerTemplate. transferTokens(.....)`
- `handlerTemplate` is an ERC20 transfer handler, and handles the token transfer.
- This fails because `mockToken` is mocked to do so.

As much as this works, the incorrect step is that you use a real transfer handler to handle the transfer, and not a mock transfer. Swap never actually touches a token itself, and so I believe you shouldn't need to mock the token at all - you only have to mock calls that Swap directly interacts with itself.

Instead you should mock:
- Mock Contracts: `mockRegistry`, `mockHandler`
- `mockRegistry. getHandler()` to return `mockHandler.address`
- `mockHandler.transferTokens` to return `false` (i.e. the transfer failed`

Reply:
That was what was breaking for me earlier. I'll try again though to see if I can get the return on the delegate call to line up.